server/src/main/java/org/elasticsearch/action/search/TransportSearchAction.java

```diff
@@ -466,6 +466,7 @@ static void ccsRemoteReduce(
         ActionListener<SearchResponse> listener,
         BiConsumer<SearchRequest, ActionListener<SearchResponse>> localSearchConsumer
     ) {
+        final var remoteClientResponseExecutor = threadPool.executor(ThreadPool.Names.SEARCH_COORDINATION);
```

Comment:
I wonder if we should use the `SEARCH` pool instead, since that is the pool other work here is spawned on.

I think the coordination pool is mainly for the node level can-match/fieldcaps actions.

Might not matter a whole lot, but I'd rather induce any rejections due to this on the `SEARCH` thread pool than the `SEARCH_COORDINATOR` pool.

Reply:
I will revert this to SAME to make progress here, and raise it with the search folks to follow-up. I think what we're doing here is more like coordination work so enqueueing it behind some IO-heavy search activity might do bad things for performance, but there's definitely tradeoffs either way.